stratum/lib/security/credentials_manager.cc

```diff
@@ -1,130 +1,161 @@
+    // Check if certificate files exist/accessible
+    std::ifstream ifile1, ifile2, ifile3;
+    ifile1.open(FLAGS_server_key_file);
+    ifile2.open(FLAGS_server_cert_file);
+    ifile3.open(FLAGS_ca_cert_file);
+    if(ifile1 && ifile2 && ifile3) {
+      auto certificate_provider =
+          std::make_shared<FileWatcherCertificateProvider>(
+              FLAGS_server_key_file, FLAGS_server_cert_file, FLAGS_ca_cert_file,
+              kFileRefreshIntervalSeconds);
+      auto tls_opts =
+          std::make_shared<TlsServerCredentialsOptions>(certificate_provider);
+      tls_opts->set_cert_request_type(GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE);
+      tls_opts->watch_root_certs();
+      tls_opts->watch_identity_key_cert_pairs();
+      server_credentials_ = TlsServerCredentials(*tls_opts);
+    } else {
+      LOG(WARNING) << "Key files provided, but cannot open. Unable to initiate server_credentials.";
+      server_credentials_ = nullptr;
     }
-    credentials_reload_interface_ =
-        std::make_shared<CredentialsReloadInterface>(
-            pem_root_certs_, server_private_key_, server_cert_);
-    auto credential_reload_config = std::make_shared<TlsCredentialReloadConfig>(
-        credentials_reload_interface_);
-
-    tls_opts_ = std::make_shared<TlsCredentialsOptions>(
-        GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE, GRPC_TLS_SERVER_VERIFICATION,
-        nullptr, credential_reload_config, nullptr);
-    server_credentials_ = TlsServerCredentials(*tls_opts_);
   }
-  return ::util::OkStatus();
-}
-
-::util::Status CredentialsManager::LoadNewCredential(
-    const std::string root_certs, const std::string cert_chain,
-    const std::string private_key) {
-  return credentials_reload_interface_->LoadNewCredential(
-      root_certs, cert_chain, private_key);
-}
 
-CredentialsReloadInterface::CredentialsReloadInterface(
-    std::string pem_root_certs, std::string server_private_key,
-    std::string server_cert)
-    : reload_credential_(true),
-      pem_root_certs_(pem_root_certs),
-      server_private_key_(server_private_key),
-      server_cert_(server_cert) {}
-
-int CredentialsReloadInterface::Schedule(TlsCredentialReloadArg* arg) {
-  absl::WriterMutexLock l(&credential_lock_);
-  if (arg == nullptr) {
-    arg->set_status(GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_FAIL);
-    return 1;
-  }
-  if (!reload_credential_) {
-    arg->set_status(GRPC_SSL_CERTIFICATE_CONFIG_RELOAD_UNCHANGED);
-    return 0;
+  // Client credentials.
+  if (FLAGS_ca_cert_file.empty() && FLAGS_client_key_file.empty() &&
+      FLAGS_client_cert_file.empty()) {
+    if (secure_only) {
+      LOG(WARNING) << "No crt/key files provided, cannot initiate gRPC server credentials";
+    } else {
+      client_credentials_ = ::grpc::InsecureChannelCredentials();
+      LOG(WARNING) << "No key files provided, using insecure client credentials!";
+    }
+  } else {
+    // Check if certificate files exist/accessible
```

Comment:
Introduced by me: Lines 102-107 checking if the files exist before accessing them for credential validation